src/pages/library/datasetBuilder/DatasetBuilder.ts

```diff
@@ -382,6 +385,86 @@ export const ValuesSelector = ({
   });
 };
 
+interface RequestAccessModalProps {
+  cohorts: Cohort[];
+  conceptSets: ConceptSet[];
+  valuesSets: HeaderAndValues<DatasetBuilderValue>[];
+  onDismiss: () => void;
+}
+
+const RequestAccessModal = (props: RequestAccessModalProps) => {
+  const { onDismiss, cohorts, conceptSets, valuesSets } = props;
+  const [name, setName] = useState('');
+  const [researchPurposeStatement, setResearchPurposeStatement] = useState('');
+
+  const required = { presence: { allowEmpty: false } };
+  const errors = validate({ name, researchPurposeStatement }, { name: required, researchPurposeStatement: required });
+
+  const nameId = _.uniqueId('');
+  const researchPurposeId = _.uniqueId('');
+
+  return h(
+    Modal,
+    {
+      title: 'Requesting access',
+      showX: false,
+      onDismiss,
+      okButton: h(
+        ButtonPrimary,
+        {
+          disabled: errors,
+          tooltip: errors && Utils.summarizeErrors(errors),
+          onClick: async () => {
+            await DatasetBuilder().requestAccess({
+              name,
+              researchPurposeStatement,
+              datasetRequest: {
+                cohorts,
+                conceptSets,
+                valuesSets: _.map((valuesSet) => ({ domain: valuesSet.header, values: valuesSet.values }), valuesSets),
+              },
+            });
+            onDismiss();
+          },
+        },
+        ['Request access']
+      ),
+    },
+    [
+      div([
+        div([
+          "A request of the dataset created will be generated and may take up to 72 hours for approval. Once approved you'll be notified by email. We'll send you a copy of this request",
```

Comment:
Curious about where the 72 hours comes from -- might this make more sense as "3 business days"?

Reply:
72 hours came from the mocks, to be honest I am not super sure where it comes from. I can forward that question on to Lou when he is back.